### PR TITLE
jolokia: add some error message context

### PR DESCRIPTION
**Pull request by ssoroka**
related to https://github.com/influxdata/telegraf/issues/7812

### CODE REVIEW

File: plugins/inputs/jolokia2/gatherer.go [hunk 1]
```diff
@@ -88,8 +88,8 @@ func (g *Gatherer) generatePoints(metric Metric, responses []ReadResponse) ([]po
 		case 404:
 			continue
 		default:
-			errors = append(errors, fmt.Errorf("Unexpected status in response from target %s: %d",
-				response.RequestTarget, response.Status))
+			errors = append(errors, fmt.Errorf("Unexpected status in response from target %s (%q): %d",
+				response.RequestTarget, response.RequestMbean, response.Status))
 			continue
 		}
```
Comment (razzrsi) [inline plugins/inputs/jolokia2/gatherer.go:95]: How to use this code in my server where telegraf is installed. Please suggest.

Comment (ssoroka) [reply plugins/inputs/jolokia2/gatherer.go:95]: Download a nightly build and test it out. https://github.com/influxdata/telegraf#nightly-builds

Comment (razzrsi) [reply plugins/inputs/jolokia2/gatherer.go:95]: I am getting below error with nightly build log. I don't understand what's wrong in this configuration during collection of undertow subsystem. Except **undertow** all other subsystems able to collect data from target server by jolokia plugin. Please help!!

 ### 2020-08-13T10:43:10Z E! [inputs.jolokia2_agent] Error in plugin: Unexpected status in response from target  ("jboss.as:http-listener=*,server=*,subsystem=undertow"): 400

Comment (ssoroka) [reply plugins/inputs/jolokia2/gatherer.go:95]: a 400 error means "bad request". This means the server couldn't understand what you're requesting, and I would suspect the mbean request. Maybe that the server doesn't understand `*`, or that it's not escaped correctly by the plugin.

I don't know why you're not seeing any output. You could try viewing it before it goes to the InfluxDB server like this:
```
[[outputs.file]]
  files = ["stdout"]
```

Comment (razzrsi) [reply plugins/inputs/jolokia2/gatherer.go:95]: It couldn't helped me as well. I didn't find any information regarding undertow subsystems for "connectors_http" in output file. Even also no information regarding such in telegraf log file. Please enhance the log by adding few contexts, why it is not collecting any data in the case of no error from the plugin site. Please Help!!!!

For the meanwhile I am looking for * symbol issue from server end.

**Comment by himuura** [comment]
Think i might have found the bug. Notice the * on the server= and http(s)-listener:
[[inputs.jolokia2_agent.metric]]
 name     = "connectors_http"
 mbean    = "jboss.as:https-listener=*,server=*,subsystem=undertow"
 paths    = ["bytesReceived","bytesSent","errorCount","requestCount"]
 tag_keys = ["server","https-listener"]

 [[inputs.jolokia2_agent.metric]]
 name     = "connectors_http"
 mbean    = "jboss.as:http-listener=*,server=*,subsystem=undertow"
 paths    = ["bytesReceived","bytesSent","errorCount","requestCount"]
 tag_keys = ["server","http-listener"]

In my case, adding the * started pouring data onto my grafana dashboard. I can now see connection statistics on Grafana.

**Pull request merged**